tests/test_causal_graph.py

```diff
@@ -538,6 +538,25 @@ def test_adjacency_consistency(self):
 
         self.assertEqual(CausalGraph.from_adjacency_matrix(*cg.to_numpy()), cg)
 
+    def test_copy(self):
+        # test copy preserves the correct class
+        ts_cg = TimeSeriesCausalGraph()
+        ts_cg.add_edge('a', 'b', meta={'foo': 'bar'})
+        ts_cg.add_edge('a lag(n=1)', 'b', meta={'test': 'test'})
+        ts_cg.add_edge('b lag(n=1)', 'b')
+
+        ts_cg_copy = ts_cg.copy()
+        self.assertIsInstance(ts_cg_copy, TimeSeriesCausalGraph)
+
+        self.assertTrue(ts_cg_copy.__eq__(ts_cg, deep=True))
+
+        # now without copying meta
+        ts_cg_copy_no_meta = ts_cg.copy(include_meta=False)
+        self.assertIsInstance(ts_cg_copy_no_meta, TimeSeriesCausalGraph)
+
+        self.assertTrue(ts_cg_copy_no_meta.__eq__(ts_cg, deep=False))
```

Comment:
This is the same as `self.assertEqual(ts_cg, ts_cg_copy_no_meta)` and can also do `self.assertEqual(ts_cg_copy_no_meta, ts_cg_copy)`